src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxDocumentConverterBase.cs

```diff
@@ -222,13 +222,20 @@ public IndexFieldsMapping GetKnownFields()
                     BlittableJsonReaderObjectsListForEnumerableScope = new(InitialSizeOfEnumerableBuffer);
                 }
 
-                var enumerableScope = new EnumerableWriterScope(StringsListForEnumerableScope, LongsListForEnumerableScope, DoublesListForEnumerableScope, BlittableJsonReaderObjectsListForEnumerableScope, _allocator);
+                bool canFinishEnumerableWriting = scope is EnumerableWriterScope;
+                var enumerableScope = canFinishEnumerableWriting
+                    ? (EnumerableWriterScope)scope
+                    : new EnumerableWriterScope(StringsListForEnumerableScope, LongsListForEnumerableScope, DoublesListForEnumerableScope, BlittableJsonReaderObjectsListForEnumerableScope, _allocator);
```

Comment:
We can do that with an `if(scope is not EnumerableWriterScope enumerableScope) { enumerableScope = new EnumerableWriterScope(StringsListForEnumerableSc... `, no?

Reply:
Yes, changed.